### PR TITLE
docs: agent config (merge after #1888)

### PR DESCRIPTION
**Pull request by chronark**
- **feat: replace redis with serf**
- **docs: add agent configuration**

### CODE REVIEW

**Comment by chronark** [comment]
ah crap, I didn’t start fresh from main..

File: apps/docs/contributing/services/agent/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+The agent is deployed as a cluster of machines to fly.io. Each machine is identical and capable of serving any incoming request.
+
+We run a gossip network between all machines for service discovery and failure detection.
+The gossip network will also be used for broadcasting configuration changes and cache invalidations.
+
+On top of the gossip layer, sits a consistent hash ring for sharding any kind of data across the cluster.
+
+<Warning>
+TODO: andreas - add more details about the architecture
```
Comment (perkinsjr) [inline apps/docs/contributing/services/agent/architecture.mdx:13]: Can we change this Warning to something that isn't a todo for yourself. 

"Details of architecture coming soon"

File: apps/docs/contributing/services/agent/configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+The agent is configured via a JSON configuration file, which is loaded at startup using the `--config=<path>` flag.
+
+Inside the config file, you may use `${ENV_NAME}` to reference environment variables. The agent will replace these references with the actual environment variable values at runtime. See example below.
+
+## $schema
+
+As the agent is in active development, the configuration spec might change frequently and this document could be outdated. 
+
+<Warning>
```
Comment (perkinsjr) [inline apps/docs/contributing/services/agent/configuration.mdx:14]: This should be an info

**Review by perkinsjr — CHANGES_REQUESTED** [review]
Lint docs, see my comments

**Comment by chronark** [comment]
fair, I’ll change

**Review by perkinsjr — APPROVED** [review]

**Pull request merged**